### PR TITLE
ddl: limit the count of getting ddlhistory jobs

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/55711

Problem Summary:
If user want to get all of DDL-history jobs, the tidb-server has the risk of OOM because of large mount of DDL-history jobs.

### What changed and how does it work?
1. Limit the count of jobs when getting the ddl history jobs.
2. The parameter 'limit' should be large than 1 and less than or equal to 1024.
3. If use default 'limit', user can get the 1024 ddl history jobs by default.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

####  Manual test steps
1. create 5k table in database with the 'sysbench' tool.
3. query the mount of ddl-history jobs, it contains 6052 ddl history jobs.
```
mysql> select count(*) from mysql.tidb_ddl_history;
+----------+
| count(*) |
+----------+
|     6052 |
+----------+
1 row in set (0.00 sec)
```
4. run below command
```
curl -X POST http://172.16.6.95:11080/ddl/history
```
 and the it gets the 1024 ddl history jobs by default.

5. run below command, it gets the fault messages when limit < 0 or limit >= 1025
```
# curl -X POST http://172.16.6.95:11080/ddl/history -d "limit=0"
ddl history limit must be greater than 0 and less than or equal to 1024

# curl -X POST http://172.16.6.95:11080/ddl/history -d "limit=1025"
ddl history limit must be greater than 0 and less than or equal to 1024

curl -X POST http://172.16.6.95:11080/ddl/history -d "limit=1024" > output2.log
```

4. run below command, it gets 1024 ddl history jobs with the parameter 'limit'.
```
curl -X POST http://172.16.6.95:11080/ddl/history -d "limit=1024" > output2.log
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 5357k    0 5357k  100    10  22.0M     42 --:--:-- --:--:-- --:--:-- 22.0M
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Limit the count of jobs when getting the ddl history jobs.
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]
> and the it gets the 1024 ddl history jobs by default.

Maybe we should notify other team about this API change

File: pkg/ddl/ddl_history.go [hunk 1]
```diff
@@ -150,9 +152,15 @@ func GetAllHistoryDDLJobs(m *meta.Meta) ([]*model.Job, error) {
 func ScanHistoryDDLJobs(m *meta.Meta, startJobID int64, limit int) ([]*model.Job, error) {
 	var iter meta.LastJobIterator
 	var err error
+
 	if startJobID == 0 {
+		// if 'start_job_id' == 0 and 'limit' == 0(default value), get the last 1024 ddl history job by defaultly.
+		if limit == 0 {
```
Comment (D3Hunter) [inline pkg/ddl/ddl_history.go:158]: this changes HTTP API, please confirm with PM @frank945946

please also update doc in here
https://github.com/pingcap/tidb/blob/master/docs/tidb_http_api.md

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: pkg/ddl/ddl_history.go (hunk 1 above)
Comment (joccau) [reply pkg/ddl/ddl_history.go:158]: > this changes HTTP API, please confirm with PM @Frank945946
> 
> please also update doc in here https://github.com/pingcap/tidb/blob/master/docs/tidb_http_api.md

Get it~

**Review by Frank945946 — APPROVED** [review]
LGTM

File: pkg/ddl/ddl_history.go (hunk 1 above)
Comment (joccau) [reply pkg/ddl/ddl_history.go:161]: I have already confirmed with @Frank945946， and confirmed with either BR team or CD team.

**Review by D3Hunter — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**